Framework/LiveData/src/LoadLiveData.cpp

```diff
@@ -277,18 +272,49 @@ void LoadLiveData::addMatrixWSChunk(const std::string &algoName,
   }
 
   // Now do the main workspace
-  IAlgorithm_sptr alg = this->createChildAlgorithm(algoName);
+  IAlgorithm_sptr alg = this->createChildAlgorithm("Plus");
   alg->setProperty("LHSWorkspace", accumWS);
   alg->setProperty("RHSWorkspace", chunkWS);
   alg->setProperty("OutputWorkspace", accumWS);
   alg->execute();
+  if (!alg->isExecuted()) {
+    throw std::runtime_error("Error when calling " + alg->name() +
+                             " to add the chunk of live data. See log.");
+  }
+}
+
+//----------------------------------------------------------------------------------------------
+/**
+ * Add a matrix workspace to the accumulation workspace.
+ *
+ * @param algoName :: Name of algorithm which will be adding the workspaces.
+ * @param accumWS :: accumulation matrix workspace
+ * @param chunkWS :: processed live data chunk matrix workspace
+ */
+void LoadLiveData::addMDWSChunk(Workspace_sptr &accumWS,
+                                const Workspace_sptr &chunkWS) {
+  // Need to add chunk to ADS for MergeMD
+  std::string chunkName = "__anonymous_livedata_addmdws_" +
+                          this->getPropertyValue("OutputWorkspace");
+  AnalysisDataService::Instance().addOrReplace(chunkName, chunkWS);
+
+  std::string ws_names_to_merge = accumWS->getName();
+  ws_names_to_merge.append(", ");
+  ws_names_to_merge.append(chunkName);
+
+  IAlgorithm_sptr alg = this->createChildAlgorithm("MergeMD");
+  alg->setPropertyValue("InputWorkspaces", ws_names_to_merge);
+  alg->executeAsChildAlg();
+
+  // No longer needed in ADS
+  AnalysisDataService::Instance().remove(chunkName);
+
   if (!alg->isExecuted()) {
```

Comment:
I know you didn't add this, but it would probably be best if we propagated the original exception. That could be done by setting `Algorithms::rethrows(true)`. Not convinced we should do that as part of the PR though.

Reply:
Looking into this a bit I found that these algorithms would rethrow anyway, because they are child algorithms. So that condition and bit of code was dead anyway (every path to isExecuted == false that I could find also led to a throw); I've simply removed it.